front/pages/api/w/[wId]/data_sources/[name]/databases/csv.ts

```diff
@@ -211,7 +211,24 @@ export default withLogging(handler);
 async function rowsFromCsv(
   csv: string
 ): Promise<Result<CoreAPIDatabaseRow[], APIError>> {
-  const parser = parse(csv);
+  // Detect the delimiter.
+  const delimiters = [",", ";", "\t"];
+  let delimiter: string | undefined = undefined;
+  for (const c of csv) {
+    if (delimiters.includes(c)) {
```

Comment:
This is a bit brittle

```
"test \t a ; cool", foo
cool, bean
```

Will fail. We should look at >1 lines count the delimiters and take the one that has the max equal number of appearance?

Reply:
That is indeed a smarter way to do it 👍 . Will fix